packages/flutter_tools/lib/runner.dart

```diff
@@ -275,9 +263,51 @@ Future<File> _createLocalCrashReport(CrashDetails details) async {
 }
 
 Future<int> _exit(int code, {required ShutdownHooks shutdownHooks}) async {
-  // Prints the welcome message if needed.
+  final FirstRunMessenger messenger =
+      FirstRunMessenger(persistentToolState: globals.persistentToolState!);
+  final bool legacyAnalyticsMessageShown =
+      messenger.shouldDisplayLicenseTerms();
+
+  // Prints the welcome message if needed for legacy analytics.
   globals.flutterUsage.printWelcome();
 
+  // Ensure that the consent message has been displayed for unified analytics
+  if (globals.analytics.shouldShowMessage) {
+    globals.logger.printStatus(globals.analytics.getConsentMessage);
+
+    // Because the legacy analytics may have also sent a message,
+    // the conditional below will print additional messaging informing
+    // users that the two consent messages they are receiving is not a
+    // bug
+    if (legacyAnalyticsMessageShown) {
+      globals.logger
+          .printStatus('You have received two consent messages because\n'
+              'the flutter tool is migrating to a new system.\n'
+              'Opting out of collection will opt you out of the legacy\n'
+              'and new analytics collection system.\n'
+              'You can opt out by running `flutter --disable-telemetry`\n'
+              'or `flutter config --no-analytics\n');
+    }
```

Comment:
Thoughts on this messaging? @zanderso 

This refers to Case 1 from https://github.com/flutter/flutter/issues/124605